tests/std/tests/GH_002769_handle_deque_block_pointers/test.cpp

```diff
@@ -0,0 +1,227 @@
+    using difference_type   = ptrdiff_t;
+    using pointer           = T*;
+    using reference         = add_lvalue_reference_t<T>;
+
+    counting_ptr(nullptr_t) noexcept : counting_ptr{static_cast<T*>(nullptr)} {}
+
+    counting_ptr() noexcept : counting_ptr{nullptr} {}
+
+    counting_ptr(const counting_ptr& other) noexcept : counting_ptr{other.p_} {}
+
+    counting_ptr& operator=(const counting_ptr&) = default;
+
+    counting_ptr& operator=(nullptr_t) noexcept {
+        p_ = nullptr;
+        return *this;
+    }
+
+    ~counting_ptr() {
+        assert(fancy_counter != 0);
+        --fancy_counter;
+    }
+
+    explicit operator bool() const noexcept {
+        return p_ != nullptr;
+    }
+
+    static counting_ptr pointer_to(T& obj) noexcept {
+        return counting_ptr{addressof(obj)};
+    }
+
+    T& operator*() const noexcept {
+        return *p_;
+    }
+
+    T* operator->() const noexcept {
+        return p_;
+    }
+
+    counting_ptr& operator++() noexcept {
+        ++p_;
+        return *this;
+    }
+
+    counting_ptr operator++(int) noexcept {
+        auto tmp = *this;
+        ++p_;
+        return tmp;
+    }
+
+    counting_ptr& operator--() noexcept {
+        --p_;
+        return *this;
+    }
+
+    counting_ptr operator--(int) noexcept {
+        auto tmp = *this;
+        --p_;
+        return tmp;
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    T& operator[](I n) const noexcept {
+        return p_[n];
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    counting_ptr& operator+=(I n) noexcept {
+        p_ += n;
+        return *this;
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    counting_ptr& operator-=(I n) noexcept {
+        p_ -= n;
+        return *this;
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    friend counting_ptr operator+(const counting_ptr& p, I n) noexcept {
+        auto tmp = p;
+        tmp += n;
+        return tmp;
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    friend counting_ptr operator+(I n, const counting_ptr& p) noexcept {
+        auto tmp = p;
+        tmp += n;
+        return tmp;
+    }
+
+    template <class I = ptrdiff_t, enable_if_t<is_integral_v<I>, int> = 0>
+    friend counting_ptr operator-(const counting_ptr& p, I n) noexcept {
+        auto tmp = p;
+        tmp -= n;
+        return tmp;
+    }
+
+    friend ptrdiff_t operator-(const counting_ptr& lhs, const counting_ptr& rhs) noexcept {
+        return lhs.p_ - rhs.p_;
+    }
+
+    friend bool operator==(const counting_ptr& p, nullptr_t) noexcept {
+        return p.p_ == nullptr;
+    }
+
+#if _HAS_CXX20
+    friend bool operator==(const counting_ptr& lhs, const counting_ptr& rhs) = default;
+
+    friend auto operator<=>(const counting_ptr& lhs, const counting_ptr& rhs) = default;

```

Comment:
No change requested: In product code, we conventionally don't name parameters of defaulted functions. On the other hand, in test code we're somewhat less rigorous about conventions. On the third hand, `operator=(const counting_ptr&)` didn't name its parameter, which is an argument for being locally consistent here. On the fourth and final hand, this isn't worth resetting testing. :joy_cat: